### ISSUE TITLE
setReadHandler never called

### ISSUE THREAD

**Issue by ApplantaDev** [lxdcn/NEPacketTunnelVPNDemo#5]
I am able to read packets from tunToUDP after my session starts. However, udpToTun gets called once to initialize the setReadHandler but it never gets called again. Is their something that I have to add to make setReadHandler get called?

**Comment by davlxd** [comment]
> udpToTun gets called once to initialize the setReadHandler but it never gets called again.

Yeah, that's usually how callbacks work - you set the callback only once. The 1st parameter of setReadHandler which is a lambda gets itself set on the UDPSession, and gets invoked automatically every time there are data available.
